### PR TITLE
Unset session `started_at` in Application Server

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Unset session `started_at`

Addresses #4766

#### Changes

- Do not set session `started_at` in AS
- Unset `started_at` as part of `as-db cleanup`

#### Testing

1. Made local device
2. Temporarily allowed setting `session.started_at` in AS (by allowing field mask validation, not part of this PR)
3. Set the started at to a value: 
```
TTN_LW_NETWORK_SERVER_ENABLED=false ttn-lw-cli dev set app1 dev1 --session.started_at "2021-01-01T12:00:00Z" --session.dev_addr ABCDABCD
```
4. Decode the value from the database:
```json
{"ids":{"device_id":"dev1","application_ids":{"application_id":"app1"},"dev_eui":"1122334455667788","join_eui":"0000000000000000","dev_addr":"ABCDABCD"},"created_at":"2021-10-27T08:34:35.761063Z","updated_at":"2021-10-27T08:37:21.740897Z","session":{"dev_addr":"ABCDABCD","keys":{},"started_at":"2021-01-01T12:00:00Z"}}
```
5. Run `as-db cleanup`
6. Decode the value from the database:
```json
{"ids":{"device_id":"dev1","application_ids":{"application_id":"app1"},"dev_eui":"1122334455667788","join_eui":"0000000000000000","dev_addr":"ABCDABCD"},"created_at":"2021-10-27T08:34:35.761063Z","updated_at":"2021-10-27T08:44:47.863279Z","session":{"dev_addr":"ABCDABCD","keys":{},"started_at":"0001-01-01T00:00:00Z"}}
```
Notice that it's reset to an empty value, the best we can do now

##### Regressions

None, unless clients were relying on faulty behavior

#### Notes for Reviewers

@kschiffer I'm leaving the target issue open and will provide some additional information on what the Console should do when merging sessions.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]

**Review by pgalic96 — COMMENTED** [review]
The changes LGTM. However, I assume we will need to run this only once and then there will be no non-zero `started_at` values anymore in the registry (might be a wrong assumption). Then it shouldn't be necessary to run this periodically with the rest of the cleanup, and maybe we can hide it behind a flag (`--unset-dev-fields` or something like that).

**Comment by adriansmares** [comment]
> The changes LGTM. However, I assume we will need to run this only once and then there will be no non-zero `started_at` values anymore in the registry (might be a wrong assumption). Then it shouldn't be necessary to run this periodically with the rest of the cleanup, and maybe we can hide it behind a flag (`--unset-dev-fields` or something like that).

Or as part of a migration, albeit I do agree that we need to this would be longer if we do that.

**Comment by johanstokking** [comment]
> The changes LGTM. However, I assume we will need to run this only once and then there will be no non-zero `started_at` values anymore in the registry (might be a wrong assumption). Then it shouldn't be necessary to run this periodically with the rest of the cleanup, and maybe we can hide it behind a flag (`--unset-dev-fields` or something like that).

Best would be to have some sort of migration version of the AS end device registry.

@adriansmares don't we have something like this for NS?

**Comment by adriansmares** [comment]
> @adriansmares don't we have something like this for NS?

We do:
https://github.com/TheThingsNetwork/lorawan-stack/blob/8edf8c81c5e8e771e778b67fb8acf4696ee0205f/cmd/ttn-lw-stack/commands/ns_db.go#L468-L491

What needs to happen for that to be useful:
- Bring `as-db migrate` from TTI, make the migration there have version 1
- Mark this cleanup as version 2
- If the recorded version is lower than `asredis.SchemaVersion`, run the missing migrations in order

**Review by htdvisser — COMMENTED** [review]
Nothing to add here. I agree that this belongs in a migration, and not in the periodic cleanup command.

**Comment by johanstokking** [comment]
@adriansmares I replaced the cleanup logic by the `as-db migrate` command that looks a lot like NS's. I updated the original comment and verified that the migration produces the desired outcome.

Output:

```
$ go run ./cmd/ttn-lw-stack as-db migrate
INFO	Connecting to Redis database...
INFO	Setting schema version	{"version": 1}
DEBUG	1 keys migrated
DEBUG	No new version available
```

And again:

```
$ go run ./cmd/ttn-lw-stack as-db migrate
INFO	Connecting to Redis database...
INFO	Existing database schema version	{"version": 1}
INFO	Database schema version is already in latest version
DEBUG	No new version available
```

File: cmd/ttn-lw-stack/commands/as_db.go [hunk 1]
```diff
@@ -29,6 +33,76 @@ var (
...
+				uidRegexpStr = idRegexpStr + `\.` + idRegexpStr
+			)
+
+			uidRegexp := regexp.MustCompile(cl.Key("uid", uidRegexpStr+"$"))
+
+			if err := ttnredis.RangeRedisKeys(ctx, cl, cl.Key("*"), 1, func(k string) (bool, error) {
+				logger := logger.WithField("key", k)
+				switch {
+				case uidRegexp.MatchString(k):
+					if err := cl.Watch(ctx, func(tx *redis.Tx) error {
```
Comment (adriansmares) [inline cmd/ttn-lw-stack/commands/as_db.go]: I'd do a `LockedWatch` here, otherwise we race with the real Application Server instances.

**Review by adriansmares — APPROVED** [review]

**Pull request merged**